src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -102,17 +107,88 @@ public void reScaffoldDifferntVersions() throws Exception {
     assertTrue(diff.identical());
 
     List<MuleConfig> muleConfigs = new ArrayList<>(result.getGeneratedConfigs());
-    ScaffoldingResult secondScaffoldingResult = scaffoldApiSync(raml, ramlFolderV2, ROOT_RAML_RESOURCE_URL_V2, muleConfigs);
+    firstScaffolding(raml, ramlFolderV2, "rescaffolding-apisync-version/v2/api.xml", ROOT_RAML_RESOURCE_URL_V2, muleConfigs);
+  }
+
+  @Test
+  public void reScaffoldDifferntVersionsWithGlobal() throws Exception {
+    String raml = "api.raml";
+    String ramlFolderV1 = "src/test/resources/rescaffolding-apisync-version/v1";
+    String ramlFolderV2 = "src/test/resources/rescaffolding-apisync-version/v2";
+    String firstScaffoldingPath = "rescaffolding-apisync-version/v1/api.xml";
+    String secondScaffoldingApiPathV1 = "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v1/api.xml";
+    String secondScaffoldingGlobalPathV1 =
+        "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v1/global.xml";
+    String secondScaffoldingApiPathV2 = "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v2/api.xml";
+    String secondScaffoldingGlobalPathV2 =
+        "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v2/global.xml";
+    int expectedMuleConfigurationSize = 2;
+    XMLUnit.setIgnoreWhitespace(true);
+
+    firstScaffolding(raml, ramlFolderV1, firstScaffoldingPath, ROOT_RAML_RESOURCE_URL, null);
+
+    List<MuleConfig> muleConfigs =
+        generateSecondScaffoldingConfigurations(secondScaffoldingApiPathV1, secondScaffoldingGlobalPathV1);
+    ScaffoldingResult secondScaffoldingResult =
+        scaffoldApiSync(raml, ramlFolderV2, ROOT_RAML_RESOURCE_URL_V2, muleConfigs, expectedMuleConfigurationSize);
+    compareFiles(secondScaffoldingResult, secondScaffoldingApiPathV2);
+    compareFiles(secondScaffoldingResult, secondScaffoldingGlobalPathV2);
```

Comment:
could receive a list of files to compare

Reply:
I changed the method, now it's verifying the scaffolding was successful instead of "compareFiles".